### PR TITLE
MIME Sniffing: tests for MIME type groups

### PR DESCRIPTION
**Pull request by JKingweb**
Group names in the tests are taken verbatim from the specification, capitalization and all.

I left the tests formatted one-per-line for easier review; if it's felt this is problematically inconsistent with other JSON test data I can, of course, reformat the file. The tests fall into a few broad groups: first tests which mirror the definitions of the specification, then negative tests, then a few useful confirmations for specific MIME types, and finally some stress tests. Any test which depends on matching against an essence or subtype has an additional test with a parameter appended.

As noted in the README, since groups are not exposed by browsers, no test harness is included.

### CODE REVIEW

File: mimesniff/mime-groups/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+== MIME type groups ==
```
Comment (annevk) [inline mimesniff/mime-groups/README.md:1]: This should probably use `##` (and the other README should too).

**Review by annevk — COMMENTED** [review]
I think this is fine and I appreciate you reworking the JSON, but let's put this at mimesniff/mime-types/resources/mime-groups.json instead and merge the two README (this at the end).

MIME type groups are a subsection of MIME types and I don't think need to have their own directory for now.

**Comment by JKingweb** [comment]
> I think this is fine and I appreciate you reworking the JSON, but let's put this at mimesniff/mime-types/resources/mime-groups.json instead and merge the two README (this at the end).
> 
> MIME type groups are a subsection of MIME types and I don't think need to have their own directory for now.

I considered doing just that, but thought better of it because the other tests are for parsing, and these are not (they work on types which have been parsed). These tests also have a different format (they have no `output` key), so any existing parsing test harness which simply runs through `*.json` might choke on the added file.

If you still think they should be merged, though, I'm fine with doing so.

**Comment by annevk** [comment]
I'm still in favor of merging. I don't want to guarantee such a precise contract for a `resources/` directory. The format goes for the files specified in the README and that's it.

**Review by annevk — APPROVED** [review]
Great, thank you!

**Pull request merged**